test/postman/inference_api_test_collection.json

```diff
@@ -62,11 +62,68 @@
 				{
 					"listen": "test",
 					"script": {
-						"id": "3abf077e-ea14-4139-abd7-1c8e462af87c",
+						"id": "d6b1f2cf-6ffb-4850-b276-108f7f65fbd9",
 						"exec": [
-							"pm.test(\"Successful POST request\", function () {",
-							"    pm.expect(pm.response.code).to.be.oneOf([200]);",
-							"});"
+							"var type_response = pm.iterationData.get(\"output_type\")",
+							"if (type_response === \"binary\" || type_response === \"others\" ) {",
+							"    pm.test(\"Successful POST request binary response\", function () {",
+							"        pm.expect(pm.response.code).to.be.oneOf([200]);",
+							"    });",
+							"}",
+							"else if (type_response === \"text\") {",
+							"    pm.test(\"Test expected TEXT response\", function () {",
+							"        pm.response.to.have.body(pm.iterationData.get(\"expected\"));",
+							"    });",
+							"",
+							"}",
+							"else",
+							" if (\"json\" === \"json\") {",
+							"    pm.test(\"Test expected JSON response\", function () {",
```

Comment:
It doesn't seem to be generic as it has been written with the current expected response body i.e. for image classification. if in future new response format/json is added then this may require complete change.

Reply:
We seem to be creating a dependency between all the tests using parametrization & validation here. I would prefer that we dont add common validation between tests in a single testcase. 

This creates a tight coupling & maintenance would become very hard.

Reply:
This has been addressed now. It is optional to add response content validation and code has been refactored to address maintainability.